cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetDefinitionLoader.java

```diff
@@ -86,7 +87,15 @@ public DatasetDefinitionLoader(LocationFactory locationFactory) {
         }
         Class<?> moduleClass = ClassLoaders.loadClass(moduleMeta.getClassName(), classLoader, this);
         DatasetModule module = DatasetModules.getDatasetModule(moduleClass);
-        module.register(registry);
+        try {
+          // TODO avoid duplicate registering of the same module (a type may have multiple transitive
```

Comment:
It is more a reminder that we should do this when we redo the dataset framework. In the current framework it is not doable.